src/main/kotlin/com/dprint/formatter/DprintFormattingTask.kt

```diff
@@ -28,72 +38,143 @@ class DprintFormattingTask(
 ) {
     private var formattingIds = mutableListOf<Int>()
     private var isCancelled = false
+    private val tracer: Tracer = TelemetryManager.getInstance().getTracer(DprintScope.FormatterScope)
 
     /**
      * Used when we want to cancel a format, so that we can cancel every future in the chain.
      */
     private val allFormatFutures = mutableListOf<CompletableFuture<FormatResult>>()
 
     fun run() {
-        val content = formattingRequest.documentText
-        val ranges =
-            if (editorServiceManager.canRangeFormat()) {
-                formattingRequest.formattingRanges
-            } else {
-                mutableListOf(
-                    TextRange(0, content.length),
+        val rootSpan =
```

Comment:
Just as an FYI this class is only run for the replacement of the default IJ formatter. So when you run shift+cmd+option+L or whatever it is. 

You probably want to have something that wraps the lower level activities that are shared between this and the code which runs if the user has configured format on save through the dprint config window. I believe that is the `EditorServiceManager` or the `Process` depending on what you want to capture in OTEL.